packages/block-library/src/media-text/edit.js

```diff
@@ -56,22 +54,9 @@ const applyWidthConstraints = ( width ) =>
 const LINK_DESTINATION_MEDIA = 'media';
 const LINK_DESTINATION_ATTACHMENT = 'attachment';
 
-class MediaTextEdit extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.onSelectMedia = this.onSelectMedia.bind( this );
-		this.onWidthChange = this.onWidthChange.bind( this );
-		this.commitWidthChange = this.commitWidthChange.bind( this );
-		this.state = {
-			mediaWidth: null,
-		};
-		this.onSetHref = this.onSetHref.bind( this );
-	}
-
-	onSelectMedia( media ) {
-		const { setAttributes } = this.props;
-		const { linkDestination, href } = this.props.attributes;
+function attributesFromMedia( { attributes, setAttributes } ) {
+	return ( media ) => {
+		const { linkDestination, href } = attributes;
```

Comment:
You could condense some of the destructuring here.
```suggestion
function attributesFromMedia( { attributes: { linkDestination, href }, setAttributes } ) {
	return ( media ) => {
```